### ISSUE TITLE
Emit custom values based on other `AddEntries`

### ISSUE THREAD

**Issue by jRimbault** [rustyhorde/vergen#365]
Hi, it's fairly easy to write a new independent provider by implementing [`AddCustomEntries`](https://docs.rs/vergen/9.0.0/vergen/trait.AddCustomEntries.html), but I'd like to be able to take values emitted by other builders such that I could take those values and manipulate them in the build script and then emit my own custom environment variables.

The use case I have in mind is building a version string at build time rather that concatenating them at runtime in my crate.

**Comment by zesterer** [comment]
Agreed, it seems a little weird that there doesn't seem to be an API to ask vergen what, say, the commit hash is without scraping `stdout` after the fact.

**Comment by CraZySacX** [comment]
Let me know if `emit_and_set` doesn't satisfy your needs.   Updating the API to expose the values for the build script would be a major change but I'm willing to consider it as getting the values off of environment variables in the build script probably isn't the best pattern.   Feel free to re-open if you would like to see that happen.
